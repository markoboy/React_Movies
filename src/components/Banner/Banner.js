import PropTypes from 'prop-types';
import React from 'react';
import './Banner.scss';

const getPictureSources = (src) => (
  <source key={src.id} media={src.media} srcSet={src.srcSet} />
);

function Banner({
  children, imgSrc, imgAlt, sources
}) {
  return (
    <section className="banner">
      <picture>
        {sources.map(getPictureSources)}
        <img src={imgSrc} alt={imgAlt} />
      </picture>
      <div className="banner__content">
        <div className="wrapper">
          {children}
        </div>
      </div>
    </section>
  );
}

Banner.defaultProps = {
  sources: [],
};

Banner.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  imgAlt: PropTypes.string.isRequired,
  sources: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      media: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
    })
  ),
};

export default Banner;
