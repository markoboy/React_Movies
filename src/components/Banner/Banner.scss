@import 'essential';

.banner {
  position: relative;
  width: 100%;
  margin: 0 auto 0.8em;
  margin-top: -5em;
  overflow: hidden;
  max-height: 420px;

  img {
    filter: blur(5px);
  }

  &__content {
    position: absolute;
    top: 0;
    left: 0;
    padding: 4.5em 15px 1em;
    width: 100%;

    @include media(minM) {
      padding: 6em 6em 1em;
    }
  }

  &__title {
    font-weight: $font-weight-regular;
    font-size: 2em;
    text-transform: uppercase;
    word-spacing: 0.2em;
    margin: 0;
    margin-bottom: 1em;
  }
}
