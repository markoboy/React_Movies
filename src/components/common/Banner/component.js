import PropTypes from 'prop-types';
import React from 'react';
import Image from '../Image';
import './styles.scss';

function BannerComponent({ children, src, srcSet, sizes, alt, width, height }) {
  return (
    <section className="banner">
      <Image
        src={src}
        srcSet={srcSet}
        sizes={sizes}
        alt={alt}
        width={width}
        height={height}
        lazy={false}
      />
      <div className="banner__content">
        <div className="wrapper">{children}</div>
      </div>
    </section>
  );
}

BannerComponent.defaultProps = {
  srcSet: undefined,
  sizes: undefined,
  width: undefined,
  height: undefined,
};

BannerComponent.propTypes = {
  src: PropTypes.string.isRequired,
  srcSet: PropTypes.string,
  sizes: PropTypes.string,
  alt: PropTypes.string.isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
};

export default BannerComponent;
