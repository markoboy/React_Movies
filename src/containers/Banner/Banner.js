// https://images.unsplash.com/photo-1563381013529-1c922c80ac8d?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1901&q=80
