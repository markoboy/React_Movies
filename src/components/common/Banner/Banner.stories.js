import React from 'react';

import BannerImg from '@resources/banner.jpg';
import Banner from './index';
import Input from '../Forms/Input';

export default {
  component: Banner,
  title: 'Common/Banner',
};

const Template = (args) => <Banner {...args} />;

export const Default = Template.bind({});
Default.args = {
  src: BannerImg,
  alt: 'Banner',
};

export const WithInput = Template.bind({});
WithInput.args = {
  src: BannerImg,
  alt: 'Banner',
  children: <Input placeholder="Search" label="Find something" />,
};

export const WithTitle = Template.bind({});
WithTitle.args = {
  src: BannerImg,
  alt: 'Banner',
  children: <h1 className="banner__title">Banner Component</h1>,
};
