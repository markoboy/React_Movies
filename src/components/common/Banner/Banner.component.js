import PropTypes from 'prop-types';
import React from 'react';
import './Banner.component.scss';

const getPictureSources = (src) => (
  <source key={src.id} media={src.media} srcSet={src.srcSet} />
);

function BannerComponent({
  children, imgSrc, imgAlt, sources
}) {
  return (
    <section className="banner">
      <picture>
        {sources.map(getPictureSources)}
        <img src={imgSrc} alt={imgAlt} />
      </picture>
      <div className="banner__content">
        <div className="wrapper">
          {children}
        </div>
      </div>
    </section>
  );
}

BannerComponent.defaultProps = {
  sources: [],
};

BannerComponent.propTypes = {
  imgSrc: PropTypes.string.isRequired,
  imgAlt: PropTypes.string.isRequired,
  sources: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      media: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
    })
  ),
};

export default BannerComponent;
