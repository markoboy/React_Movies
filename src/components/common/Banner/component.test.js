import React from 'react';
import { render } from '@testing-library/react';

import Banner from './component';

describe('Banner Component', () => {
  it('renders Banner Component with provided image', () => {
    const { container } = render(<Banner imgSrc="/image-path/img.jpg" imgAlt="Testing banner component" />);

    expect(container).toMatchSnapshot();
  });
});
