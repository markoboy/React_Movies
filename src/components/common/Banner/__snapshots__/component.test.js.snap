// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component renders Banner Component with provided image 1`] = `
<div>
  <section
    class="banner"
  >
    <picture>
      <img
        alt="Testing banner component"
        src="/image-path/img.jpg"
      />
    </picture>
    <div
      class="banner__content"
    >
      <div
        class="wrapper"
      />
    </div>
  </section>
</div>
`;
